drivers/flash/flash_stm32_ospi.c

```diff
@@ -0,0 +1,1498 @@
+	dev_data->hospi.Init.FreeRunningClock = HAL_OSPI_FREERUNCLK_DISABLE;
+	dev_data->hospi.Init.ClockMode = HAL_OSPI_CLOCK_MODE_0;
+	dev_data->hospi.Init.WrapSize = HAL_OSPI_WRAP_NOT_SUPPORTED;
+	dev_data->hospi.Init.SampleShifting = HAL_OSPI_SAMPLE_SHIFTING_NONE;
+	/* STR mode else Macronix for DTR mode */
+	if (dev_cfg->data_rate == OSPI_DTR_TRANSFER) {
+		dev_data->hospi.Init.MemoryType = HAL_OSPI_MEMTYPE_MACRONIX;
+		dev_data->hospi.Init.DelayHoldQuarterCycle = HAL_OSPI_DHQC_ENABLE;
+	} else {
+		dev_data->hospi.Init.MemoryType = HAL_OSPI_MEMTYPE_MICRON;
+		dev_data->hospi.Init.DelayHoldQuarterCycle = HAL_OSPI_DHQC_DISABLE;
+	}
+	dev_data->hospi.Init.ChipSelectBoundary = 0;
+	dev_data->hospi.Init.DelayBlockBypass = HAL_OSPI_DELAY_BLOCK_USED;
+	dev_data->hospi.Init.Refresh = 0;
+
+	if (HAL_OSPI_Init(&dev_data->hospi) != HAL_OK) {
+		LOG_ERR("OSPI Init failed");
+		return -EIO;
+	}
+
+#if defined(CONFIG_SOC_SERIES_STM32U5X)
+	/* OCTOSPI I/O manager init Function */
+	OSPIM_CfgTypeDef ospi_mgr_cfg = {0};
+
+	__HAL_RCC_OSPIM_CLK_ENABLE();
+	if (dev_data->hospi.Instance == OCTOSPI1) {
+		ospi_mgr_cfg.ClkPort = 1;
+		ospi_mgr_cfg.DQSPort = 1;
+		ospi_mgr_cfg.NCSPort = 1;
+		ospi_mgr_cfg.IOLowPort = HAL_OSPIM_IOPORT_1_LOW;
+		ospi_mgr_cfg.IOHighPort = HAL_OSPIM_IOPORT_1_HIGH;
+	} else if (dev_data->hospi.Instance == OCTOSPI2) {
+		ospi_mgr_cfg.ClkPort = 2;
+		ospi_mgr_cfg.DQSPort = 2;
+		ospi_mgr_cfg.NCSPort = 2;
+		ospi_mgr_cfg.IOLowPort = HAL_OSPIM_IOPORT_2_LOW;
+		ospi_mgr_cfg.IOHighPort = HAL_OSPIM_IOPORT_2_HIGH;
+	}
+
+	ospi_mgr_cfg.Req2AckTime = 1; /* arbitrary value */
+
+	if (HAL_OSPIM_Config(&dev_data->hospi, &ospi_mgr_cfg,
+		HAL_OSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("OSPI M config failed");
+		return -EIO;
+	}
+
+	/* OCTOSPI2 delay block init Function */
+	HAL_OSPI_DLYB_CfgTypeDef ospi_delay_block_cfg = {0};
+
+	ospi_delay_block_cfg.Units = 56;
+	ospi_delay_block_cfg.PhaseSel = 2;
+	if (HAL_OSPI_DLYB_SetConfig(&dev_data->hospi, &ospi_delay_block_cfg) != HAL_OK) {
+		LOG_ERR("OSPI DelayBlock failed");
+		return -EIO;
+	}
+#endif /* CONFIG_SOC_SERIES_STM32U5X */
+
+	/* reset NOR flash memory : still with the SPI/STR config for the NOR */
+	if (stm32_ospi_mem_reset(dev) != 0) {
+		LOG_ERR("OSPI reset failed");
+		return -EIO;
+	}
+	/* check if memory is ready in the SPI/STR mode */
+	if (stm32_ospi_mem_ready(&dev_data->hospi,
+		OSPI_SPI_MODE, OSPI_STR_TRANSFER) != 0) {
+		LOG_ERR("OSPI memory not ready");
+		return -EIO;
+	}
+
+	if (stm32_ospi_config_mem(dev) != 0) {
+		LOG_ERR("OSPI mode not config'd (%u rate %u)",
+			dev_cfg->data_mode, dev_cfg->data_rate);
+		return -EIO;
+	}
+
+	/* send the instruction to read the NOR-flash ID : check access to mem. */
+	uint8_t nor_id[3];
+
+	ret = ospi_read_norid(dev, nor_id, sizeof(nor_id));
+	if (ret != 0) {
+		LOG_ERR("reading Nor-flash ID failed: %d", ret);
+		return ret;
+	}
+
+	LOG_DBG("Nor-flash ID : Manuf 0x%X, Mem Type 0x%X, Density 0x%X",
+		nor_id[2], nor_id[0], nor_id[1]);
```

Comment:
Please, could the `jedec_id` be used instead of `nor_id`?

```suggestion
	uint8_t jedec_id[3];

	ret = ospi_read_norid(dev, jedec_id, sizeof(jedec_id));
	if (ret != 0) {
		LOG_ERR("JEDEC ID read failed: %d", ret);
		return ret;
	}

	LOG_DBG("JEDEC ID : Manuf 0x%X, Mem Type 0x%X, Density 0x%X",
		nor_id[2], nor_id[0], nor_id[1]);
```

Reply:
Ok, I will remove this function entirely. It was useful to control that a NOR flash  not responding to SFDP command, was even capable (alive) to answer to NOR id command.